urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/codegen/Class.java

```diff
@@ -0,0 +1,191 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.codegen;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.*;
+
+/**
+ *
+ * @author ertri
+ */
+public class Class {
+
+    public String pack = "erilex.generated";
+    public boolean abs;
+    public String name;
+    public Type[] superType;
+    public List<String> params = new ArrayList<String>();
+    public List<Method> methods = new ArrayList<Method>();
+    public List<Variable> fields = new ArrayList<Variable>();
+    public List<Class> classes = new ArrayList<Class>();
+    public static final String PRE = "/** Code generated by EriLex */\n";
+    public boolean inner = false;
+    public boolean isStatic = false;
+    public String[] imports;
+    public String[] staticImports;
+    public boolean inter = false;
+
+    public Type getType() {
+        Type[] tparams = new Type[params.size()];
+
+        int i = 0;
+        for(String p : params) {
+            tparams[i++] = Generator.cons(p);
+        }
+        return Generator.cons(name, tparams);
+    }
+
+    public Class(String pack, boolean abs, String name, Type... superType) {
+        this.pack = pack;
+        this.abs = abs;
+        this.name = name;
+        this.superType = superType;
+    }
+
+    public Class(String pack, boolean abs, String name) {
+        this.pack = pack;
+        this.abs = abs;
+        this.name = name;
+    }
+
+    public Class(String pack, String name) {
+        this.pack = pack;
+        this.name = name;
+    }
+    public Class(String name) {
+        this.inner = true;
+        this.name = name;
+    }
+    public Class(boolean abs, String name, Type... superType) {
+        this.inner = true;
+        this.abs = abs;
+        this.name = name;
+        this.superType = superType;
+    }
+    public Class(String pack, String name, Type... superType) {
+        this.pack = pack;
+        this.name = name;
+        this.superType = superType;
+    }
+    public Class(String name, Type... superType) {
+        this.inner = true;
+        this.name = name;
+        this.superType = superType;
+    }
+
+    public void writeToFile(String path) throws IOException {
+        String[] s = pack.split(";")[0].split("\\.");
+        for(String d : s) {
+            path += File.separator+d;
+        }
+        File dir = new File(path);
+        if(!dir.exists()) {
+            dir.mkdirs();
+        }
+        File file = new File(dir, name+".java");
+        Writer w = new BufferedWriter(new FileWriter(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *w*. The resource is referenced at lines 109, 110, 111. The resource is closed at line 112. There are other execution paths that do not contain closure statements, for example, when Writer.flush throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).